build.sbt

```diff
@@ -181,3 +175,18 @@ lazy val testWorkspace = project
     scalacOptions -= "-Xlint"
   )
   .disablePlugins(ScalafixPlugin)
+
+lazy val `sbt-metals` = project
+  .settings(
+    sbtPlugin := true,
+    // we use 1.0 (instead of 1.1) to ensure compatibility with all 1.* versions
+    sbtVersion := "1.0.4",
+    crossSbtVersions := Seq("0.13.17", sbtVersion.value),
+    libraryDependencies := Seq(),
+    scalacOptions := scalacOptions.value diff {
+      Seq("-Yrangepos") ++
+        Seq("-Ywarn-unused-import").filter { _ =>
+          scalaBinaryVersion.value == "2.10"
+        }
```

Comment:
I find this whole `diff` pretty hard to read. I took me a while to understand.

Doesn't this work?

```scala
scalacOptions --= Seq("-Yrangepos", "-Ywarn-unused-import")
```

Reply:
I admit that this `diff` doesn't look nice. I wanted to remove `-Ywarn-unused-import` only on Scala 2.10, but I guess it's not important, I'll change it to `--=` 👍

Reply:
you can apply the filter on that `Seq` too, but unless it's necessary I would avoid complicating it.